src/chapter2/01-html-and-the-dom.md

```diff
@@ -0,0 +1,139 @@
+
+> We will later talk about character encodings in great detail.
+
+### The Document Object Model
+
+Once you open an HTML document in a browser, it is represented by the **Document Object Model** (DOM for short). Basically DOM is a model for documents which represent *editable logical trees*. This sounds *way* scarier than it really is. Let's explain this using an example. Looking at our HTML file, we have some HTML that looks like this:
+
+```html
+<div id="todoList">
+  <p id="task1">
+    Read the MERN book: Read and understand the MERN book.
+  </p>
+  <p id="task2">
+    Write a website: Create a new and cool website.
+  </p>
+</div>
+```
+
+Here we have a `<div>` element (which is a generic container) and two `<p>` (paragraph) elements inside that div element. The `<div>` element is the *parent* of the `<p>` elements and the `<p>` elements are the *children* of the `<div>` element. The elements form a *tree*.
+
+Consider this HTML:
+
+```html
+<div id="app">
+  <h1 id="headline">My tasks</h1>
+  <div id="todoList">
+    <p id="task1">
+      Read the MERN book: Read and understand the MERN book.
+    </p>
+    <p id="task2">
+      Write a website: Create a new and cool website.
+    </p>
+  </div>
+</div>
+```
+
+You can visualize it like this:
+
+![](images/tree.png)
+
+> As a side note, HTML elements do not need to have an ID.
+
+## Manipulating the DOM using JavaScript
+
+Whenever you write applications which have a lot of logic in the client, you will need to manipulate the DOM (i.e. add or remove items). For example, if we add a task, we might need to add some elements to the current tree representing the task. This is how the corresponding JavaScript™ code would look like:
+
+```javascript
+const taskText = 'Do something else: Just do something already.'
+const element = document.createElement('p');
+element.innerHTML = taskText;
+const todoList = document.getElementById('todoList');
+todoList.append(element);
+```
+
+This JavaScript™® code already looks cumbersome and annoying. It will only get more annoying once the logic becomes more complex. Manually manipulating the DOM in JavaScript™®℠ is generally a bad idea as it leads to unreadable and unmaintainable code. And unreadable code is almost as bad as inserting unnecessary trademark symbols into a text just for the heck of it. Your fellow developers will *really* hate you if you do that.
```

Comment:
Is this the same joke as in a previous chapter or was it in the earlier version?

Reply:
It was in an earlier version